### PR TITLE
Demonstrated more Vapor features, integrated optional database

### LINKED ISSUE #29: Add DB
Add an example of using Fluent with something like Postgresql. Perhaps a basic user system

### PR DESCRIPTION
**Pull request by NathanFlurry**
Resolves #29, but does it in a way which does not require the user to set up a database. It should work fine without a database attached. I added endpoints at `/hash`, `/board`, `/db-version`, `/magic`, and `/math`.

### CODE REVIEW

**Comment by NathanFlurry** [comment]
If someone (probably @tannernelson) could take a shot at making `/board` pretty, that'd be great. CSS and HTML will be the death of me.

File: App/Models/Post.swift [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	}
+	
+	func makeNode() throws -> Node {
+		/**
+			
+		*/
+		return try Node(node: [
+			"id": id,
+			"text": text,
+//			"user": try user().get() // TODO: Shouldn't this work?
```
Comment (tanner0101) [inline App/Models/Post.swift:34]: This should work. What was the error?

File: App/Models/User.swift [hunk 2]
```diff
@@ -1,30 +1,97 @@
...
 import Fluent
 
 final class User: Model {
     var id: Node?
-    var name: String
+    var name: String // The username belonging to this user
     
     init(name: String) {
-        self.name = name
+		/**
```
Comment (tanner0101) [inline App/Models/User.swift:10]: Looks like tabs need to be converted to spaces throughout.

File: App/main.swift [hunk 3]
```diff
@@ -40,15 +64,82 @@ drop.get("/") { request in
...
+	Vapor currently limits you to three path components.
+	When you need more than three path components,
+	consider using a group.
+*/
+drop.get("math", Int.self) { request, number in
+	return try JSON([
+		"number":      number,
+		"number * 2":  number * 2,
+		"number / 2":  number / 2,
+		"number << 2": number << 2
```
Comment (tanner0101) [inline App/main.swift:82]: Cool!

File: Config/development/mysql.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+	"__comment": "See README.md for instructions on setting up the database.",
```
Comment (tanner0101) [inline Config/development/mysql.json:2]: `//` comments should work now that we are parsing with NSJSON. Do they not? :(

**Comment by tanner0101** [comment]
Woohoo! This is awesome.

Just need to copy the `.travis.yml` from a MySQL repo.

Like here: https://github.com/vapor/mysql/blob/master/.travis.yml

**Comment by tanner0101** [comment]
done in #43

**Pull request closed without merging**